call-profile/main.go

```diff
@@ -0,0 +1,566 @@
+	"SKIPPED_SAME_LAST_REJECTED_DIFF":                               "skippedSameLastRejectedDiff",
+	"SKIPPED_SAME_LAST_PENDING_DIFF":                                "skippedSameLastPendingDiff",
+	"SKIPPED_CURRENT_USER_DATA_SAME_AS_DIFF":                        "skippedCurrentUserDataSameAsDiff",
+	"SKIPPED_OTHER_ERROR":                                           "skippedOtherError",
+	"SKIPPED_VALIDATION_ERROR":                                      "validation error",
+}
+
+/*
+ Setting Firestore Key for development/production
+*/
+func getFirestoreKey() string {
+	if os.Getenv(("environment")) == Constants["ENV_DEVELOPMENT"] {
+		return os.Getenv(Constants["FIRE_STORE_CRED"])
+	} else if os.Getenv(("environment")) == Constants["ENV_PRODUCTION"] {
+		var parameterName string = Constants["FIRE_STORE_CRED"]
+
+		sess := session.Must(session.NewSessionWithOptions(session.Options{
+			SharedConfigState: session.SharedConfigEnable,
+		}))
+
+		svc := ssm.New(sess)
+
+		results, err := svc.GetParameter(&ssm.GetParameterInput{
+			Name: &parameterName,
+		})
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+
+		return *results.Parameter.Value
+	} else {
+		return ""
+	}
+}
+
+/*
+ Utils
+*/
+
+/*
+ Function to initialize the firestore client
+*/
+func initializeFirestoreClient(ctx context.Context) (*firestore.Client, error) {
+	sa := option.WithCredentialsJSON([]byte(getFirestoreKey()))
+	app, err := firebase.NewApp(ctx, nil, sa)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := app.Firestore(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func (res Res) Validate() error {
+	return validation.ValidateStruct(&res,
+		validation.Field(&res.FirstName, validation.Required),
+		validation.Field(&res.LastName, validation.Required),
+		validation.Field(&res.Phone, validation.Required, is.Digit),
+		validation.Field(&res.Email, validation.Required, is.Email),
+		validation.Field(&res.YOE, validation.Min(0)),
+		validation.Field(&res.Company, validation.Required),
+		validation.Field(&res.Designation, validation.Required),
+		validation.Field(&res.GithubId, validation.Required),
+		validation.Field(&res.LinkedIn, validation.Required),
+		validation.Field(&res.Website, is.URL))
+}
+
+/*
+ MODELS
+*/
+
+/*
+ Logs the health of the user's service
+*/
+func logHealth(client *firestore.Client, ctx context.Context, userId string, isServiceRunning bool, sessionId string) {
+	newLog := Log{
+		Type:      Constants["PROFILE_SERVICE_HEALTH"],
+		Timestamp: time.Now(),
+		Meta: map[string]interface{}{
+			"userId":    userId,
+			"sessionId": sessionId,
+		},
+		Body: map[string]interface{}{
+			"userId":         userId,
+			"serviceRunning": isServiceRunning,
+		},
+	}
+	client.Collection("logs").Add(ctx, newLog)
+}
+
+/*
+ Logs the status of the user's profileDiff
+*/
+func logProfileSkipped(client *firestore.Client, ctx context.Context, userId string, reason string, sessionId string) {
```

Comment:
in Body field of the Log struct should be a map of strings to interface{} values. However, the value of the reason key is a string, not an interface{} value. To fix this bug, you need to change the line of code to the following:

newLog.Body = map[string]interface{}{
		"userId": userId,
		"reason": interface{}(reason),
	}

Reply:
Ok, so I did this to use a map to work with dynamic JSON and the interface accepts the dynamic value. So I think it should work this way also and it is working this way.
Correct me please if I am wrong.